test/cppsuite/test_harness/workload/database_operation.h

```diff
@@ -165,35 +167,45 @@ class database_operation {
             cursors.push_back(cursor);
         }
 
-        cpt = 0;
-        /* Walk each cursor. */
-        for (const auto &it : cursors) {
-            collection_name = collection_names[cpt];
-            /* Walk each key. */
-            for (keys_iterator_t iter_key = context.get_collection_keys_begin(collection_name);
-                 iter_key != context.get_collection_keys_end(collection_name); ++iter_key) {
-                /* Do not process removed keys. */
-                if (!iter_key->second.exists)
-                    continue;
-
+        /*
+         * Update each collection while the test is running.
+         */
+        i = 0;
+        while (context.is_running() && !collection_names.empty()) {
+            if (i >= collection_names.size())
+                i = 0;
+            ret = cursors[i]->next(cursors[i]);
+            /* If we have reached the end of the collection, reset. */
+            if (ret == WT_NOTFOUND) {
+                cursors[i]->reset(cursors[i]);
+                ++i;
+                continue;
+            }
+            /* Stop updating in case of an error. */
```

Comment:
Move this comment inside the if to avoid the weird formatting, additionally we should testutil_die("reason", ret) here. If we can't traverse a cursor to the next entry that is unexpected and should abort / fail the test.